### ISSUE TITLE
CPU Spike 90% with Vsync 

### ISSUE THREAD

**Issue by howdev** [linuxmint/cinnamon#9621]
```
 * Cinnamon version (cinnamon 4.4)
   - Please specify if you are using the daily builds PPA
     (https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds).
 * Distribution - (Mint 19.3)
 * Graphics Nvidia 440.59
 * 64 bit

**Issue**
When Chrome or any Chromium based browser running with hardware acceleration on
All Vsync methods None, Fallback, Throttle, Presentation Time, causes Xorg and Cinnamon to 90% CPU usage while moving any application window for more than 2 seconds. None vsync has less CPU spike.

Tested in XFCE. does not have the issue. 


**Steps to reproduce**
Use any Chrome, Opera, Chromium or Brave browser with hardware acceleration on
continuously drag any windows for more than 2 seconds, the CPU will start to spike up to 90%, monitor with top command


**Expected behaviour**
Cpu is very low usage, should be under 10%

**Comment by howdev** [comment]
Also happens with Cinnamon 4.2

**Comment by sanawu** [comment]
Mint 20 64-bit
Cinnamon 4.6.4
VSync set to presentation time
NV drivers 390.138 
Brave Hardware acceleration enbled

CPU normal

**Comment by howdev** [comment]
@sanawu  Did you move any windows? if idle nothing will happen. you move any other app window, not just browser window, Xorg will jump to 90%

**Comment by sanawu** [comment]
> @sanawu Did you move any windows? if idle nothing will happen. you move any other app window for more than 2 second, just drag around the screen, Xorg will jump to 90%
> 
> if Vsync None; Xorg 60%, Cinnamon 30%
> if Vsync Presentation; Xorg 90%, Cinnamon 9%
> 
> XFCE; Xorg 15%

Yes I moved it about a good bit for a good while, nothing dissruptive happened, Linux still works, desktop functioning, CPU not spiking massively

**Comment by Philipp91** [comment]
> When Chrome or any Chromium based browser running with hardware acceleration on
All Vsync methods None, Fallback, Throttle, Presentation Time, causes Xorg and Cinnamon to 90% CPU usage while moving any application window for more than 2 seconds.

I have roughly the same issue. Though I move/change window _contents_ (e.g. playing a video or scrolling a webpage) instead of the whole window. (When I move a window, there's also some `cinnamon` CPU usage, but I won't complain because I'll allow my window manager to consume CPU when I'm doing something with a window -- and that's a much less frequent use case than playing a video.) But indeed, with Firefox everything's fine. OTOH with Joplin (which is a React Native app) I can also reproduce the high `cinnamon` CPU load while scrolling.

> None vsync has less CPU spike.

I've observed the same. I also tried various other ways of disabling VSync (on my second monitor), but it didn't help.

In addition to the above, **I _only_ observe this when a Cinnamon panel exists on my monitor with VSync** (see also [here](https://github.com/linuxmint/cinnamon/issues/8460#issuecomment-826315068) and [here](https://forums.linuxmint.com/viewtopic.php?p=2007375#p2007375)). My other monitor (which is my primary monitor) does not have/support/use VSync (it's connected through HDMI, maybe that's why), and Cinnamon panels there don't hurt. If you only have a VSync-ed monitor, it might be inconvenient but possible to try out if removing the Cinnamon panel on that monitor improves the situation (though you might have to reconfigure it from scratch later?).

**Comment by howdev** [comment]
I upgraded to Mint 20 and issue stays the same. I tried XFCE and KDE don't have this problem.

**Comment by Philipp91** [comment]
Is there any way to debug what's going on here? Is there a way to monitor CPU time of Cinnamon's sub-threads? Is there a way to see when or how often a rerender of Cinnamon's UI is triggered (e.g. through VSync)?

**Comment by Philipp91** [comment]
This reproduces with generic and lowlatency kernel, and with Nvidia drivers 460.80 and 465.27, but **not** with the nouveau (1:1.0.16-1) driver. Maybe because that one doesn't support VSync at all. Sadly it also can't control the graphics card's fan at all and is rumored to be much less stable, so I'm disinclined to use it.
